Assets/Plugins/iOS/Shopify/WebCheckoutSession.swift

```diff
@@ -29,14 +29,15 @@ import WebKit
 
 private var activeWebPaySession: WebCheckoutSession?
 
+enum WebUIState: String {
+    case willPresent        = "willPresent"
+    case willDismiss        = "willDismiss"
+    case didDismiss         = "didDismiss"
+    case loadingThankYou    = "loadingThankYou"
+}
+
 protocol WebCheckoutDelegate: class {
-    func willPresentWebCheckout()
-    func didPresentWebCheckout()
-    
-    func didDismissWebCheckout()
-    func willDismissWebCheckout()
-    
-    func didCompletePurchase()
+    func onChangedWebUIState(newState: WebUIState, receivedUnityResponseCallback: UnityMessage.MessageCompletion?)
```

Comment:
Nit: Seems very Javascripty to me. A more conventional name would be something like:
```swift
func checkout(checkout: WebCheckoutSession, didChange state: State, completion: UnityMessage.MessageCompletion?)
```